### ISSUE TITLE
ddev list should show health of site based on all containers, not just web

### ISSUE THREAD

**Issue by rfay** [drud/ddev#318]
## What happened (or feature request):

In testing site restart (and docker restart) both @tannerjfco and I noticed the mysql container going into a restart loop (a different error) but really, really not healthy. But ddev list shows the site as "running" even though it's completely broken.

## What you expected to happen:

The state of key containers should be considered before declaring that a site is "running" on startup and in list. Currently it only takes into account the web container.

## How to reproduce it (as minimally and precisely as possible):

Right now today, just do a `ddev restart`, and then look at `docker ps` to see the restarts happening due to a mysql bug. But a list will tell you everything is OK.

## Related source links or issues:

**Comment by rickmanelius** [comment]
Questions:
- What specific test would you perform to ensure mysql was up? Attempt a mysql login for each active app?
- What word would you use if something wasn't healthy? It technically wouldn't be stopped or running. Would you need/want to state additional information to the end-user to resolve?

**Comment by rfay** [comment]
Docker has a HEALTHCHECK command that we use for both the web and router containers; we should be able to roll one for mysql container. Just a trivial query is good enough. 

"not functional" for broken container?

**Comment by rickmanelius** [comment]
@rfay Noted. Then sounds like a low LOE and therefore worth pursuing. Thoughts:

* It seems like it would make sense to focus on just the web and mysql server. However, when someone uses a custom container, would it be appropriate to require that in the check?
* "not functional" is accurate, but kinda long and we've been trying to save horizontal space. What about "failing"? If you don't think that is as accurate, then I'm happy to go with "not functional".
* As an end-user, it would be helpful to know what to do in these situations. Do I go visit a link to read on how to address? Do I just run a start/stop? Perhaps if any site has a "not functional" status, we append at the end of the list. "One of your site's is in a 'not functional' status, which means one of your containers failed a docker health check. We recommend tryings a ddev restart. If the problem persists, please file an issue at https://github.com/drud/ddev/issues/."

If we're all aligned on this, I'm happy to make this actionable and put in this release.

**Comment by rfay** [comment]
FYI I think @tannerjfco is already working on this and nearly has a PR ready. Could be wrong.

"Borked" would probably be a short column that would be understood.

For user *reaction* to nonfunctional, we do need to add ideas to the FAQ that I don't think exists yet (or troubleshooting section, whatever).

What I currently do first is
* ddev start (even if already running) - almost completely safe, but might update/recreate containers, so could break something like the situation where a user has installed an alternate drush version.
* ddev restart
* ddev rm -y && ddev start
* ddev rm -y && rm -r .ddev && ddev config; ddev start

Mostly my perception is that I generally need those approaches for problems we've created with the upgrade path or switching ddev versions.

**Comment by rickmanelius** [comment]
Stubbing out an FAQ section in our docs would work for me. Then we can link to it and then describe the item in more detail as well as the steps to take, kindof like what you describe in your above comment.

**Comment by rickmanelius** [comment]
TL;DR. I think we're good to go on this. Summarizing:

- We use the docker health check.
- We pick one of the words to describe this new state.
- If one or more sites is in this failing state, we provide a link to suggested fixes in our docs.
